### ISSUE TITLE
@ApiParam ignores certain properties

### ISSUE THREAD

**Issue by milosonator** [springfox/springfox#2107]
I have a Spring Boot Project with `springfox-swagger2` `2.7.0` And I have the following controller:

```java
@Api(tags = { "Some" }, description = "CRUD for Some Stuff")
@RestController
@RequestMapping(path = "/some")
public class SomeController {

  @ApiOperation(value = "Get some")
  @GetMapping(value = "{someId}", produces = MediaType.APPLICATION_JSON_VALUE)
  public Response getSomeById(@PathVariable("someId") Id someId) {
    return ...;
  }
...
}
```
I want to control what is displayed in the docs by annotating the `Id` class, and this is working only for some parts of the annotation, but not all.
The `Id` class (which has a registered converter from String to Id): 

```java
public class Id {

  @ApiParam(value = "This is the description", defaultValue = "1f1f1f", required = true, name = "someId", type = "string")
  private final Long id;

  public Id(Long id) {
    this.id = id;
  }

  public Long getId() {
    return id;
  }
}
```

Now the swagger JSON returned looks as follows:
```json
"parameters":[{
  "name":"id",
  "in":"query",
  "description":"This is the description",
  "required":true,
  "type":"integer",
  "default":"1f1f1f",
  "format":"int64"
}]
```

My question (or possibly bug report) is: why are some parts of the `@ApiParam` annotation used (like `value`, `defaultValue` and `required`), but others aren't, like `name` and `type`?
Why does it seem like I cannot change the `name` or `type` here?
For my particular use case, the latter is the one I would want to change to 'string'.

**Comment by dilipkrish** [comment]
I believe its a bug that its not using name correctly. however, it will use the correct type as described by the type system, unless you use the fully qualified type name `java.lang.String`

**Comment by milosonator** [comment]
I've tried the fully qualified name as well, to no avail.

**Comment by dilipkrish** [comment]
Will look into it.. seems like a bug

**Comment by milosonator** [comment]
If anyone is interested in a workaround:
https://stackoverflow.com/questions/47317562/swagger-apiparam-ignores-certain-properties

**Comment by abhikt48** [comment]
Hi Team,
I am also facing same issue. 

`@GET
@RequestMapping(value = "/validate/login", method = RequestMethod.GET, produces = "application/json", headers={"password", "userName"} )
public String validateLogin(HttpServletRequest request,
		@ApiParam(value = "UserName", required = true) @HeaderParam(value = "userName") String userName,
		@ApiParam(value = "Password", required = true, format="password") @HeaderParam(value = "password") String password) {
	
	
	return "Success"; 
	
}`
**Generated JSON Specs** -- "format: password"  is missing in generated specs. 
`{
	"name": "password",
	"in": "header",
	"description": "Password",
	"required": true,
	"type": "string"
}`

@milosonator  - I tried to work on alternative which you suggested to use OverrideSwaggerApiParamBuilder. I implemented this but **no one method of this class is invoking**. 
Please find attached "OverrideSwaggerApiParamBuilder" and "MainApplication" class. 

Could you please suggest me, why no one methods of "OverrideSwaggerApiParamBuilder" invoking?

[MainApplication.txt](https://github.com/springfox/springfox/files/1505479/MainApplication.txt)
[OverrideSwaggerApiParamBuilder.txt](https://github.com/springfox/springfox/files/1505480/OverrideSwaggerApiParamBuilder.txt)

I am using below dependency. 
<dependency>
	<groupId>io.springfox</groupId>
	<artifactId>springfox-swagger-ui</artifactId>
	<version>2.7.0</version>
</dependency>
<dependency>
	<groupId>io.springfox</groupId>
	<artifactId>springfox-spi</artifactId>
	<version>2.7.0</version>
</dependency>
<dependency>
	<groupId>io.springfox</groupId>
	<artifactId>springfox-swagger2</artifactId>
	<version>2.7.0</version>
</dependency>

**Comment by dilipkrish** [comment]
@abhikt48 this seems like a new feature request. Would you mind creating a new issue for this?

**Comment by dilipkrish** [comment]
Closed via 1606caf4a421470ccb9b7592b465979dcb4c5ce1

**Comment by milosonator** [comment]
:smile:
